src/lib/storage/table.cpp

```diff
@@ -436,8 +440,28 @@ void Table::add_soft_key_constraint(const TableKeyConstraint& table_key_constrai
   _table_key_constraints.insert(table_key_constraint);
 }
 
-const TableKeyConstraints& Table::soft_key_constraints() const {
-  return _table_key_constraints;
+void Table::delete_key_constraint(const TableKeyConstraint& constraint) {
+  _table_key_constraints.erase(constraint);
+}
+
+bool Table::constraint_guaranteed_to_be_valid(const TableKeyConstraint& table_key_constraint) const {
```

Comment:
I see why this is a member of `Table`, but I'd prefer this to be a helper function in a separate file taking the table and the constraint as arguments.